src/NSwag.AspNetCore/IDocumentProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IDocumentProvider.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.ApiDescription
```

Comment:
Is it a good idea to add the Microsoft namespace to this package? This may confuse people. Or will this be moved into another library eventually?

Reply:
This namespace is a marker for use in the tool we'll ship. It can take many forms i.e. is up to the document generator provider. So, linking it to the `Microsoft.Extensions.ApiDescription.Design` package, at least by inference, makes sense.

Reply:
Resolve as it is internal only now.